refactor: template 수정 기능

daamont opened the pull request:
# refactor: 템플릿 검증 시스템 연결 및 수정 기능 개선

- 템플릿 생성 부분 리팩토링 완료 시 그에 맞춰서 수정 부분 추가 리팩토링이 필요해 보임
- 머지 및 자잘한거 수정중...

---

## ✅ PR 종류
- [ ] feat (새로운 기능 추가)
- [ ] fix (버그 수정)
- [x] refactor (코드 리팩토링)
- [ ] docs (문서 추가 및 수정)
- [ ] style (비즈니스 로직 영향 없는 변경)
- [ ] test (테스트 코드 관련 작업)
- [ ] chore (빌드/설정 등 기타 변경)

---

## 📝 작업 내용

### 기존 기능 개선
- **변수 표시 토글**: 실시간 변수 하이라이트 기능
- **채팅 UI**: 자동 스크롤 및 사용자 경험 개선
- **템플릿 제목 관리**: 템플릿 수정 시 제목 자동 생성 및 처리
- **API 요청 구조**: 템플릿 제목을 포함한 완전한 요청 데이터 구조

### 🗑️ 제거된 기능
- **사용자 직접 수정 기능**

---

## 🔍 변경 이유

1. 직접 수정 기능을 제거하고 채팅 기반 수정으로 통일
2. **API 안정성**: 템플릿 제목 처리 및 요청 구조 개선

### TemplateService에서 Account 엔티티 대신 accountId를 파라미터로 받도록 변경
- 보안 강화: 외부에서 임의의 Account 객체가 주입되는 위험 제거
- 서비스 경계 단순화: Service 레이어는 id만 받고, 엔티티 조회 책임을 내부에서 수행
- JPA 영속성 관리 일관성: getReferenceById 등을 활용해 안전하게 영속성 컨텍스트에서 관리
-> 그에 따른 templateController에서 getId로 정보 가져오기

---

## 🔄 데이터 흐름

1. **생성 시 가져오는 데이터:**
   ```typescript
   {
     templateContent: string,        // 사용자가 작성한 템플릿 내용
     variables: Record<string, any>, // 템플릿에 사용된 변수들
     category: string,              // 템플릿 카테고리
     userMessage: string,           // 사용자 원본 요청 메시지
     templateTitle: string,         // 템플릿 제목
     variableList: VariableDto[]    // 변수 정보를 DTO 배열로 변환
   }
   ```

2. **백엔드에서 AI 서버로 전송하는 검증 요청:**
   ```java
   {
     "template": {
       "channel": "alimtalk",
       "body": "템플릿 내용",
       "variables": {변수들},
       "category": "marketing",
       "title": "템플릿 제목"
     },
     "user_input": "사용자 원본 메시지"
   }
   ```

---

## ✅ 체크리스트
- [x] 코드 동작 확인
- [x] 관련 테스트 작성/수정
- [ ] 문서 업데이트
- [ ] 리뷰 반영

daamont committed:
refactor: 탬플릿 수정 부분 개선 및 변수 토클 수정
- 변수 표시 토글 로직 개선 및 기본값 설정
- 제출 시 템플릿이 DB에 저장되도록 수정

daamont committed:
feat: 변수 직접 수정 기능
- 수정 버튼 클릭 시 변수가 수정가능하도록 변환

daamont committed:
fix: 템플릿 수정 API 에러 수정 및 사용자 수정 기능 개선
- TemplateModificationRequest에 current_template_title 필드 추가
- 프론트엔드에서 템플릿 제목을 API 요청에 포함하도록 수정
- 채팅 수정과 사용자 수정 버튼 하나만 사용하도록 비활성화 기능 추가

daamont committed:
feat: 템플릿 검증 시스템 연결 및 사용자 직접 수정 기능 제거
- 채팅 UI 자동 스크롤 추가
- 템플릿 제목 처리 및 API 요청 구조 개선
- 사용자 직접 수정 관련 코드 삭제

choichanwoo001 commented on package-lock.json:
```diff
@@ -0,0 +1,323 @@
+{
+  "name": "final_6team_pls_jober",
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {
+    "": {
+      "dependencies": {
+        "vue": "^3.5.21",
```
코드가.............................................. package-lock.json, package 내용이 바뀐것 같거든요?

그라고 이거는 애초에 올라가있어야 했던건데 왜 지금 4000줄 가량이 올라간건지 모르겠어요............

daamont replied:
저도 왜 4000줄이 넘게 나온지는 모르겠지만,, 파일 수정사항은 몇줄 안되더라구요,,,
package-lock.json, package 내용 디벨롭에서 가져왔는데 그렇게 가져온게 문제일까여?

choichanwoo001 replied:
흠...................................................... 그냥 pull 받았다는 말씀이시져?

daamont replied:
json 파일은 복붙하고 나머지는 pull 받아왔숩니당,,,

choichanwoo001 replied:
lock 파일, package.json파일 내용이 서로 바뀐 것 같습니다! 
두개 반대로 복붙하고 다시 올려주세요

그리고 두개 파일 모두 front 디렉토리 바로 아래 넣어주셔야합니다.

choichanwoo001 commented on .gitignore:
```diff
@@ -36,7 +36,7 @@ ai/chroma_db/
 ai/.env
 
 # ---------- Node (front) ----------
-front/node_modules/
+node_modules/
 front/dist/
 front/.vite/
```
이거 이렇게 수정한게.. 혹시 왜일까요? root 기준으로는 front/node_modules가 맞긴한데

daamont replied:
저의 경로는 front/에 위치하지 않아서 수정했습니다

choichanwoo001 replied:
<img width="217" height="62" alt="image" src="https://github.com/user-attachments/assets/3f05f1d4-3c26-4d8d-83d7-3dfc7f016535" />
front 폴더 아래 node_modules가 없다는거에요?

daamont replied:
넹 전체 폴더 아래 node_modules이 생겼숨다

choichanwoo001 commented on ai/routers/ai_routes.py:
```diff
@@ -49,21 +49,24 @@ class QuestionAnswerRequest(BaseModel):
     category: str
     user_message: str
-    model: Optional[str] = "gpt-4o-mini"
+    model: Optional[str] = "gpt-5"
 
```
모델 다시 4o-mini로 바꿔주세요!

choichanwoo001 commented on ai/templateEngine/message_analyzer.py:
```diff
@@ -4,8 +4,8 @@
-from ai.services.openai_service import OpenAIService
-from ai.templateEngine.prompts.message_analyzer_prompts import TypePromptBuilder, CategoryPromptBuilder, FieldsPromptBuilder
+from services.openai_service import OpenAIService
+from templateEngine.prompts.message_analyzer_prompts import TypePromptBuilder, CategoryPromptBuilder, FieldsPromptBuilder
```
이거 경로 수정되는거 혹시 ai 폴더를 root로 기준잡고 하셔서 이렇게 수정한건가요?
아니면 그냥 이게 맞는건가요

daamont replied:
ai. 붙여서하면 경로 오류가 나서 제거했습니다

daamont committed:
Merge branch 'develop' into feature/template-update

# Conflicts:
#	front/src/api/index.ts
#	front/src/views/TemplateCreateView.vue

choichanwoo001 commented on front/src/components/KakaoPreviewComponent.vue:
```diff
@@ -7,15 +7,13 @@
       <div class="kakao-content">
         <div class="kakao-title">
           <span>쿠폰 발급 안내</span>
```
이 하드코딩은 아직 수정하지 않은 이유가 있을까요?

daamont replied:
생성쪽 파트에서 진행하는 거라 생각해서 아직 수정안했는데 이 부분도 제가 수정할 부분이었을까요?

choichanwoo001 replied:
아ㅏ 맞네요 sorry ~

choichanwoo001 reviewed (changes requested):
수고하셨습니다! comment들 확인해주셔요

choichanwoo001 commented:
<img width="1201" height="283" alt="image" src="https://github.com/user-attachments/assets/78389d8c-b842-4bf7-b888-15b4d527fcc3" />
충돌도 해결해주세요!  

develop은 templateCreateView 삭제하고 모달로 바꿔가지고 .. 그냥 수민님꺼 반영 안하고 뭘 수정하려고 했는지 알려주시면 이거 수정할때 참고해서 반영하겠습니다.

choichanwoo001 commented:
<img width="1062" height="282" alt="image" src="https://github.com/user-attachments/assets/29ab27cd-8407-44d8-ada4-ef52b672cccb" />
그리고 이 부분이 정확이 프론트에서 수집되는게 맞을까요? 

수집되는건 사용자 원본 데이터이고 나머지는 결국 back으로 넘어가기 전까지 준비되는 항목들 아닌가요?
다시 말하면 사용자가 front 딴에서는 입력하는 데이터가 아닌 것 같아 여쭤봅니다

daamont committed:
fix: 템플릿 수정 API 응답에서 설명 텍스트 제거
- AI 서버에 템플릿 추출 로직 추가하여 순수한 템플릿만 반환
- 프롬프트 개선으로 AI가 설명을 포함하지 않도록 유도
- 변수 표시 형식 개선

daamont commented:
프론트에서 수집한 데이터 -> 생성에서 수정할 때 가져오는 데이터로 정정
TemplateCreateView 삭제하고 모달로 바꿔가지고 ...........................................
여기선 크게 수정한 내용이 없어서 삭제하고 push할까요?
-> 채팅 ui 수정한거랑 수정 채팅 시 프롬프트, 그리고 템플릿 보여줄 때 쓸데없는 단어나 이런거 제외하고 올리는 ai/routers 코드 참고하면 될 것 같네요,,,,,

daamont committed:
fix: gpt 모델명 변경
- gpt-5에서 gpt-4o-mini로 변경
 Changes to be committed:
	modified:   ai/routers/ai_routes.py

daamont committed:
chore: 불필요한 파일 제거

choichanwoo001 reviewed (approved):
수고하셨습니다! 충돌 해결하고 진행하시면 좋을 것 같아여

daamont committed:
Merge branch 'develop' into feature/template-update

choichanwoo001 committed:
Merge branch 'develop' into feature/template-update

daamont committed:
refactor: 템플릿 수정 결과 화면 리팩토링 및 코드 정리
- UI/UX 개선 및 컴포넌트 간소화
- 백엔드 AI 서비스 로직 추가
- 불필요한 파일 제거 및 코드 최적화

daamont committed:
Merge branch 'develop' into feature/template-update

daamont committed:
refactor: 불필요한 변수 제거 및 python ci 해결
- variables제거하고 variableList 통일

daamont committed:
Merge branch 'develop' into feature/template-update

daamont committed:
fix: TemplateController 컴파일 오류 수정

choichanwoo001 reviewed (approved):
수고하셨습니다!!!!!

daamont committed:
fix: 삭제된 코드 복원
 Changes to be committed:
	modified:   src/main/java/com/example/controller/TemplateController.java
	modified:   src/main/java/com/example/service/TemplateService.java

choichanwoo001 committed:
Merge branch 'develop' into feature/template-update
